docs/root/configuration/security/cert_provider.rst

```diff
@@ -0,0 +1,14 @@
+.. _config_certificate_provider:
```

Comment:
I have a few high level comments that we need to work through (apologize in advance but this is the first time this type of PR has happened and there are some things we are going to need to sort out).
1) At a very high level, this documentation goes into the Envoy documentation. As such, these docs are not useful to Envoy users as there are no such extensions available (nor is the extension mechanism itself implemented), and Envoy users are not specifically looking for information on gRPC.
2) *However*, I recognize that we are entering the multi-dataplane world and are going to need to figure out how to handle situations like this. My general thoughts on this are that for proto documentation (things like not implemented, client specific docs, etc.), we are going to need to have an overlay where we can have client specific documentation. @htuch has already been working on this for security stuff and I think this would work for client specific docs also. For non-client specific docs like this, it's not clear to me what to do. I think it would be reasonable to put this into the extension portion of the Envoy docs, but it's still really going into an Envoy specific doc location.

Given that these are Envoy docs and not proto docs, my feeling is to drop this PR for now, especially given that the the extension mechanism itself is not implemented yet in Envoy. I assume you have gRPC documentation where you could put this type of documentation?

cc @envoyproxy/api-shepherds @envoyproxy/maintainers for comment.

Reply:
Thanks @mattklein123. The reason I opened this PR is because I am writing up a doc (a  proposal for the new gRPC xDS security feature to be precise) where I needed to reference the new security protos and their description. When I looked up the current Configuration reference  for security the only thing described is SDS and I thought adding the new stuff for Certificate Provider might not be a bad idea. I understand Envoy itself doesn't support Certificate Providers and that's why I did't consider adding to "Introduction » Architecture overview » Security" but adding to "Configuration reference » Security" might be okay.

But I am okay with dropping this PR and I'll find alternative ways to reference this stuff from my doc. I'll close it after your response.

Reply:
My preference would be to close for now, but I wanted to be clear that we are committed to figuring out the multi-dataplane doc story, so I didn't want to appear to be dismissing your effort here.

Reply:
So this is the gRPC proposal doc https://github.com/grpc/proposal/pull/184 for which I wanted to provide some reference material in this doc.